OpenSim/Moco/MocoGoal/MocoContactImpulseTrackingGoal.cpp

```diff
@@ -88,21 +88,16 @@ void MocoImpulseTrackingGoal::addScaleFactor(const std::string& name,
         "Contact group associated with external force '{}' not found.",
         externalForceName);
 
-    // Check that the index is in the correct range.
-    OPENSIM_THROW_IF_FRMOBJ((index < 0) || (index > 2), Exception,
-        "Expected marker scale factor index to be in the range [0, 2], "
-        "but received '{}'.", index);
-
     // Update the scale factor map so we can retrieve the correct MocoScaleFactor
     // for this contact group during initialization.
-    std::pair<std::string, int> key(externalForceName, index);
+    std::pair<std::string, int> key(externalForceName, impulse_axis);
     m_scaleFactorMap[key] = name;
 
     // Append the scale factor to the MocoGoal.
     appendScaleFactor(MocoScaleFactor(name, bounds));
 }
 
-void MocoImpulseTrackingGoal::initializeOnModelImpl(const Model& model) const {
+void MocoContactImpulseTrackingGoal::initializeOnModelImpl(const Model& model) const {
```

Comment:
Until we support multiple integrals/impulses, we should add a check that the `contact_groups` property has length == 1.  You can add a comment to the documentation in the header file noting that the goal currently only supports tracking one impulse for one foot at a time.